UI/obs-app.cpp

```diff
@@ -3176,6 +3217,47 @@ static void upgrade_settings(void)
 	os_closedir(dir);
 }
 
+static const char *sentinelContents = "This file intentionally left blank.";
+
+static void check_safe_mode_sentinel(void)
+{
+#ifndef NDEBUG
+	// No auto safe mode in debug builds
+	return;
+#else
+	if (disable_shutdown_check)
+		return;
+
+	char path[512];
+	int pathlen = GetConfigPath(path, 512, "obs-studio/safe_mode");
+
+	if (pathlen <= 0)
+		return;
+
+	if (!os_file_exists(path)) {
+		os_quick_write_utf8_file(path, sentinelContents,
```

Comment:
Is `os_quick_write_utf8_file` guaranteed to flush the buffer to disk immediately? Because otherwise this might not do anything.

Reply:
It does flush the file: https://github.com/obsproject/obs-studio/blob/b31344d82ec60608e03b9648f5ff96fa486d5c9a/libobs/util/platform.c#LL280C7-L280C7